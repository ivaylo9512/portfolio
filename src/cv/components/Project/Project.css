.project{
    position: relative;
    display: inline-block;
    margin-bottom: 5%;
}
.clip{
    height: 100%;
    width: 100%;
    position: absolute;
    left: 0;
    top: 0;
    overflow: hidden;
    border-radius: 0.5vw;
    fill: black;
}
.project-rect{
    display: block;
    height: 11vw;
    width: 20vw;
    fill: white;
}
.project-rect rect{
    filter: url(#project-shadow-top) url(#project-shadow);
}