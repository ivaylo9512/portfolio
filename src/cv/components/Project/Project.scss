.project{
    position: relative;
    display: inline-block;
    margin-bottom: 5%;
    width: 46%;
    color: #312929;

    .content{
        position: absolute;
        height: 100%;
        width: 100%;
        padding: 2% 0 5% 4%;
        box-sizing: border-box;
        justify-content: space-between;
        display: flex;
        flex-direction: column;
        font-weight: bold;
        text-shadow: 0 0 0.1vw;

        svg{
            margin-right: 2%;
        }

        .name{
            margin: 0;
            font-weight: bold;
            font-size: 1.2vw;
        }

        a{
            font-weight: bold;
            font-size: 1.1vw;
            text-decoration: underline;
            color: #312929;

            &:hover,
            &:focus{
                text-decoration: none;
                outline: none;
                transform: scale(1.1);
            }
        }

        .github-container{
            display: flex;
            align-items: center;
            height: 25%;

            svg{
                height: 50%;
                margin-right: 2%;
            }
        }

        .info{
            display: flex;
            align-items: center;
            height: 31%;
            width: 80%;
            font-size: 0.9vw;

            .circle{
                height: 36%;
            }
        }
    }

    .clip{
        height: 100%;
        width: 100%;
        position: absolute;
        left: 0;
        top: 0;
        overflow: hidden;
        fill: black;
        pointer-events: none;
        border-radius: calc(0.55vw * 0.69 * 1);
    }

    .rect{
        display: block;
        width: 100%;
        fill: white;
    
        rect{
            filter: url(#project-shadow);
        }
    }
}