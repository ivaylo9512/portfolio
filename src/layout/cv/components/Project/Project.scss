.left-panel .project{
    position: relative;
    margin: 4% 0;
    width: 75%;
    color: #312929;
    font-family: Myriad Pro Regular;
    margin: 2.5% 0;
    padding-bottom: 30%;
    box-shadow: 0 0 3px rgba(0, 0, 0, 0.5);
    border-radius: 10px;
    transition: padding 2s, transform 0s;
    background: hsl(0, 13%, 97%);

    &:hover{
        transform: scale(1.02);
    }

    .content{
        a,
        .github-container,
        .info{
            opacity: 1;
            transition: opacity 1s;
        }
    }

    &.hidden{
        .content{
            a,
            .github-container,
            .info{
                opacity: 0;
                pointer-events: none;
            }
        }
    }
    .content{
        position: absolute;
        height: 100%;
        width: 100%;
        padding: 2% 0 5% 4%;
        box-sizing: border-box;
        justify-content: space-between;
        display: flex;
        flex-direction: column;
        font-weight: bold;

        svg{
            margin-right: 2%;
        }

        .name{
            margin: 0;
            font-size: 4.2vw;
        }

        a{
            font-size: 2.1vw;
            text-decoration: underline;
            color: #312929;

            &:hover,
            &:focus{
                text-decoration: none;
                outline: none;
                transform: scale(1.1);
            }
        }

        .github-container{
            display: flex;
            align-items: center;
            height: 25%;

            svg{
                height: 50%;
                margin-right: 2%;
            }
        }

        .info{
            display: flex;
            align-items: center;
            height: 31%;
            width: 80%;
            font-size: 1.9vw;

            .circle{
                height: 36%;
                filter: url(#shape-shadow);
           }
        }
    }

    .clip{
        height: 100%;
        width: 100%;
        position: absolute;
        left: 0;
        top: 0;
        overflow: hidden;
        pointer-events: none;
        border-radius: calc(0.55vw * 0.69 * 1);
    }

    .rect{
        display: block;
        width: 100%;
        fill: white;
    
        rect{
            filter: url(#shape-shadow);
        }
    }

    @media only screen and (min-width : 700px) and (orientation : landscape) {
        width: 51%;
        margin: 2.5% 0;

        .content{
            .name{
                font-size: 1.1vw;
            }

            a{
                font-size: 0.9vw;
            }

            .info{
                font-size: 0.8vw;
            }
        }
    }

    @media only screen and (min-width : 896px) and (max-width : 1100px ) and (orientation : landscape) {
        .content{
            .name{
                font-size: 1.8vw;
            }

            a{
                font-size: 1.3vw;
            }

            .info{
                font-size: 1.1vw;
            }
        }
    }

    @media only screen and (min-width: 700px) and (max-width : 896px) and (orientation : landscape) { 
        .content{
            .name{
                font-size: 2vw;
            }

            a{
                font-size: 1.5vw;
            }

            .info{
                font-size: 1.3vw;
            }
        }
    }
}
