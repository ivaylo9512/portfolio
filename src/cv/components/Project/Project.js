import Circle from 'cv/icons/Circle';
import './Project.scss';
import Github from 'svgIcons/Github';
import Rect from 'cv/icons/Rect';
import { useSelector } from 'react-redux';
import { isProjectsSection } from 'app/slices/animationsSlice';

const Project = ({ height, width, name, repo, info }) => {
    const isProfile = useSelector(isProjectsSection);

    return(
        <div className='project'>
            <div className='content'>
                <p className='name'>{name}</p>
                <div className='github-container'>
                    <Github />
                    <a tabIndex={isProfile ? '-1' : undefined} href={`https://github.com/ivaylo9512/${repo}`}>Github</a>
                </div>
                <div className='info'>
                    <Circle />
                    <p>{info}</p>
                </div>
            </div>
            <div className='clip'>
                <Circle />
            </div>
            <Rect height={height} width={width} />
        </div>
    )
}
export default Project