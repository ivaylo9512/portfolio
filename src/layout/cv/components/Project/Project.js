import Circle from '@svgIcons/Circle';
import './Project.scss';
import Github from 'svgIcons/Github';
import Rect from '@svgIcons/Rect';
import TabLink from '@cv/components/TabLink/TabLink';

const Project = ({ height, width, name, repo, info }) => {
    return(
        <div className='project'>
            <div className='content'>
                <p className='name'>{name}</p>
                <div className='github-container'>
                    <Github />
                    <TabLink shouldMatch={true} href={`https://github.com/ivaylo9512/${repo}`}>
                        Github
                    </TabLink>
                </div>
                <div className='info'>
                    <Circle />
                    <p>{info}</p>
                </div>
            </div>
            <div className='clip'>
                <Circle />
            </div>
            <Rect height={height} width={width} />
        </div>
    )
}
export default Project