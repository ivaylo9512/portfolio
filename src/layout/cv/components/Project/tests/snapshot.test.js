import Project from '../Project';
import { shallow } from 'enzyme';

describe('Project snapshot tests', () => {
    const createWrapper = () => shallow(
        <Project />
    )

    it('should match snapshot', () => {
        const wrapper = createWrapper();
     
        expect(wrapper).toMatchSnapshot();
    })
});