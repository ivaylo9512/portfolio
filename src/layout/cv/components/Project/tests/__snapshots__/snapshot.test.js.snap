// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project snapshot tests should match snapshot 1`] = `
<div
  className="project"
>
  <div
    className="content"
  >
    <p
      className="name"
    />
    <div
      className="github-container"
    >
      <Github />
      <TabLink
        href="https://github.com/ivaylo9512/undefined"
        shouldMatch={true}
      >
        Github
      </TabLink>
    </div>
    <div
      className="info"
    >
      <Circle />
      <p />
    </div>
  </div>
  <div
    className="clip"
  >
    <Circle />
  </div>
  <Rect />
</div>
`;
